ui/src/variables/utils/resolveSelectedValue.ts

```diff
@@ -17,10 +15,25 @@ export const resolveSelectedKey = (
 }
 
 export const resolveSelectedValue = (
-  values: VariableMapObject,
+  values: string[],
   selectedKey: string,
   defaultSelection?: string
 ): string => {
+  if (
+    values &&
+    typeof values === 'object' &&
+    Object.prototype.toString.call(values) === '[object Array]'
+  ) {
+    if (values.includes(selectedKey)) {
+      return selectedKey
+    }
+
+    if (values.includes(defaultSelection)) {
+      return defaultSelection
+    }
+
+    return values[0]
+  }
```

Comment:
Is this entire if statement a particular variable condition (like a map)? If so, we could export it into it's own helper function to make things a little clearer

Reply:
i have no idea why we would add a helper function to a helper function of a helper function if not for a case made around reusability, but we also don't need to reuse this at all outside of this context. the "if array, do this, else do this" seems pretty clear?

Reply:
Totally. So then maybe we can refactoring the original condition to be:

```
Array.isArray(values)
```

rather than:

```
 if (
    values &&
    typeof values === 'object' &&
    Object.prototype.toString.call(values) === '[object Array]'
  )

Reply:
👍